src/WCSimTrajectory.cc

```diff
@@ -226,10 +261,40 @@ void WCSimTrajectory::AppendStep(const G4Step* aStep)
       std::vector<G4float> bPs(3);
       bPs[0] = track->GetPosition().x(); bPs[1] = track->GetPosition().y(); bPs[2] = track->GetPosition().z();
       AddBoundaryPoint(bPs, track->GetKineticEnergy(), track->GetGlobalTime(), ty);
-      // G4cout<<"Step point "<<track->GetCurrentStepNumber () <<" "<<track->GetPosition().x()<<" "<<track->GetPosition().y()<<" "<<track->GetPosition().z()<<
-      //    " "<<track->GetKineticEnergy()<<" "<<thePrePV->GetName()<<" "<<thePostPV->GetName()<<G4endl;
     }
   }
+
+#ifdef WCSIM_SAVE_PHOTON_HISTORY
+  // Add photon history
+  if (PDGEncoding==0)
+  {
+    const G4VProcess* pds = thePostPoint->GetProcessDefinedStep();
+    //G4cout<<"Having optical photon in AppendStep "<<pds->GetProcessName()<<G4endl;
+    if ( pds->GetProcessType() == fOptical )
+    {
+      if ( pds->GetProcessSubType() == fOpRayleigh || pds->GetProcessSubType() == fOpMieHG )
+      {
+        AddPhotonScatter(1);
```

Comment:
Naive question, but would it ever be useful to know how many of each scattering process (rather than the combined total)?
Now would be the perfect time to split the numbers up